code/mob/living/critter.dm

```diff
@@ -1305,6 +1305,7 @@ ADMIN_INTERACT_PROCS(/mob/living/critter, proc/modify_health)
 			if (isintangible(C)) continue
 			if (isdead(C)) continue
 			if (istype(C, src.type)) continue
+			if (src.faction != null && (C.faction == src.faction)) continue //Checks if they are in the same faction
```

Comment:
If you're doing bitflags for faction alignment, you should do `C.faction & src.faction != 0` instead. That way if they share any factions they are friends, and if not, enemies.